fast_llm/data/dataset/gpt/sampled.py

```
@@ -432,10 +431,8 @@ def _lazy_load(self):
 
     def _load_yaml_data(self, data: dict[str, typing.Any]) -> None:
         self._documents_per_epoch = data["dataset"]["documents_per_epoch"]
-        if unshuffled_tokens := data.get("unshuffled_tokens") is not None:
-            self._unshuffled_tokens = unshuffled_tokens
-        else:
-            self._unshuffled_tokens = data["unshuffled_epochs"] * data["dataset"]["tokens_per_epoch"]
+        if "unshuffled_tokens" in data:
+            self._unshuffled_tokens = data["unshuffled_tokens"]
```

should we care about backwards compatibility?

We could throw an error if `unshuffled_tokens` is not present in `loaded_yaml_data`. And speaking of backwards compatibility, I think we should also add `truncate_documents: True` in `loaded_yaml_data` if not present, else it will fail loading previous datasets.

---

The `if` was because of the method being called before `"unshuffled_tokens"` is added to data, I reorganized things so it's not needed.
We don't really care about backward compatibility for dataset cache, but I added a simple one.